pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -178,55 +234,78 @@ func (p *ecrProvider) Enabled() bool {
 		Name: "k8s/logger",
 		Fn:   awsHandlerLogger,
 	})
-	p.getter = getter
+	return getter, nil
+}
 
-	return true
+// GetTokenGetterForRegion gets the token getter for the requested region. If it
+// doesn't exist, it creates a new ECR token getter
+func (f *ecrTokenGetterFactory) GetTokenGetterForRegion(region string) (tokenGetter, error) {
+	f.mutex.Lock()
+	defer f.mutex.Unlock()
+
+	if getter, ok := f.cache[region]; ok {
```

Comment:
It seems we can introduce RWLock and use read lock here, write lock for line 253.